test/libsolidity/semanticTests/various/selfdestruct_post_cancun_multiple_beneficiaries.sol

```diff
@@ -0,0 +1,60 @@
+contract C {
+    constructor() payable {}
+    function terminate(address _beneficiary) public {
+        selfdestruct(payable(_beneficiary));
+        assert(false);
+    }
+}
+
+contract D {
+    address account1 = payable(0x5e17b14ADd6c386305A32928F985b29bbA34Eff5);
+    address account2 = payable(0x9d83e140330758a8fFD07F8Bd73e86ebcA8a5692);
+    C public c;
+
+    constructor() payable {}
+
+    function deploy() public payable {
+        c = new C{value: 1 ether}();
+    }
+
+    function terminate(address _beneficiary) public {
+        c.terminate(_beneficiary);
+    }
+
+    function test_deploy_and_terminate_twice() public {
+        deploy();
+        terminate(account1);
+        terminate(account2);
+    }
+
+    function exists() public view returns (bool) {
+        return address(c).code.length != 0;
+    }
+}
+// ====
+// EVMVersion: >=cancun
+// ----
+// constructor(), 2 ether ->
+// balance: 0x5e17b14ADd6c386305A32928F985b29bbA34Eff5 -> 0
+// balance: 0x9d83e140330758a8fFD07F8Bd73e86ebcA8a5692 -> 0
+// balance -> 2000000000000000000
+// exists() -> false
+// test_deploy_and_terminate_twice() ->
+// exists() -> false
+// balance: 0x5e17b14ADd6c386305A32928F985b29bbA34Eff5 -> 1000000000000000000
+// balance: 0x9d83e140330758a8fFD07F8Bd73e86ebcA8a5692 -> 0
+// balance -> 1000000000000000000
+// deploy() ->
+// exists() -> true
+// balance: 0x5e17b14ADd6c386305A32928F985b29bbA34Eff5 -> 1000000000000000000
+// balance: 0x9d83e140330758a8fFD07F8Bd73e86ebcA8a5692 -> 0
+// balance -> 0
+// terminate(address): 0x5e17b14ADd6c386305A32928F985b29bbA34Eff5 ->
+// balance: 0x5e17b14ADd6c386305A32928F985b29bbA34Eff5 -> 2000000000000000000
+// balance: 0x9d83e140330758a8fFD07F8Bd73e86ebcA8a5692 -> 0
+// balance -> 0
+// terminate(address): 0x9d83e140330758a8fFD07F8Bd73e86ebcA8a5692 ->
+// balance: 0x5e17b14ADd6c386305A32928F985b29bbA34Eff5 -> 2000000000000000000
+// balance: 0x9d83e140330758a8fFD07F8Bd73e86ebcA8a5692 -> 0
+// balance -> 0
+// exists() -> true
```

Comment:
I guess this is testing the same case in multiple transactions? I don't think that's necessary. It clearly cannot do anything wrong because after the first `terminate()` balance is 0.

I'd remove these extra expectations because that way you could also cover all EVM versions with the same test, without splitting it.

Reply:
It is basically what is different from the test `pre_cancun`. Here, the second call to `terminate` for `account2` will not revert but will be a no-op since the contract has no balance.

Reply:
Yeah, but checking that not really the point of this test. It would be simpler if we didn't do it.

Reply:
oh ok.

Reply:
I thought other tests checked that. Even if not, they could.